Covid19Radar/Covid19Radar/Services/TestNativeImplementation.cs

```diff
@@ -4,79 +4,198 @@
+                => Regex.IsMatch(CdnUrlBase, @"^https://.*\..*\..*/$");
+
+        public bool IsDirectInput()
+        => Regex.IsMatch(CdnUrlBase, @"^(\d+,)+\d+,*$");
+
+
+        private ushort NumberEndofSentence(string url)
+        {
+            Match match = Regex.Match(url, @"(?<d>\d+)$");
+            ushort number = 0;
+            if (match.Success)
+            {
+                number = Convert.ToUInt16(match.Groups["d"].Value);
+            }
+            return (number);
+        }
+        public List<string> GetCreatedTimes()
+            => CdnUrlBase.Split(",").ToList();
+        public ushort GetTekListDataType()
+        => NumberEndofSentence(CdnUrlBase);
+        public string[] GetApiUrlSegment()
+        {
+            // "url/api" -> { "url/api", "", "" }
+            // "url/base/api/register1/diagnosis2" -> { "url/base/api", "/register1", "/diagnosis2" } 
+            // "url/api1/r1/d2" -> { "url/api1", "/r1", "/d2" } 
+            // "url/api1/d2/r1" -> { "url/api1", "/r1", "/d2" } 
+            var url = ApiUrlBase;
+            var r = new Regex("/r(egister)?[0-9]+");
+            var d = new Regex("/d(iagnosis)?[0-9]+");
+            var urlRegister = r.Match(url).Value;
+            url = r.Replace(url, "");
+            var urlDiagnosis = d.Match(url).Value;
+            url = d.Replace(url, "");
+            var urlApi = url;
+            return (new string[] { urlApi, urlRegister, urlDiagnosis });
+        }
+        public ushort GetDiagnosisDataType()
+        => NumberEndofSentence(GetApiUrlSegment()[2]);
+        public ushort GetRegisterDataType()
+        => NumberEndofSentence(GetApiUrlSegment()[1]);
+        public ushort GetApiDataType()
+        => NumberEndofSentence(GetApiUrlSegment()[0]);
+        public bool IsDirectInputApi()
+        => Regex.IsMatch(GetApiUrlSegment()[0], @"^(\d+,)+\d+,?$");
+        public List<string> GetApiStrings()
+            => GetApiUrlSegment()[0].Split(",").ToList();
+    }
+    public class TestNativeImplementation : INativeImplementation
+    {
+        static readonly Random random = new Random();
+        private readonly MockCommonUtils mockCommonUtils;
+
+        Task WaitRandom()
+            => Task.Delay(random.Next(100, 2500));
+
+        public async Task StartAsync()
+        {
+            await WaitRandom();
+            Preferences.Set("fake_enabled", true);
+        }
+
+        public async Task StopAsync()
+        {
+            await WaitRandom();
+            Preferences.Set("fake_enabled", false);
+        }
+
+        public async Task<bool> IsEnabledAsync()
+        {
+            await WaitRandom();
+            return Preferences.Get("fake_enabled", true);
+        }
+
+        public async Task<IEnumerable<TemporaryExposureKey>> GetSelfTemporaryExposureKeysAsync()
+        {
+            var keys = new List<TemporaryExposureKey>();
+
+            for (var i = 1; i < 14; i++)
+                keys.Add(GenerateRandomKey(i));
```

Comment:
ここは for 文に { } でブロックを付けてください。
（ブロックがないと気づかずに、あとから行を増やしたときに1行しか繰り返されなかったという苦い思い出があるので…）

あと、`14`がマジックナンバーになってしまっているので、定数`DAY_OF_TEK_STORED`として定義すると後から見てわかりやすいと思います。

```suggestion
            for (var i = 1; i <= DAY_OF_TEK_STORED; i++)
            {
                keys.Add(GenerateRandomKey(i));
            }
```